REST usage fails with native when e.g. ContainerResponseFilter is used

### Describe the bug

ClassLoader cannot find the REST endpoint class

```
Caused by: java.lang.NoSuchMethodException: com.alesj.qcl.app.HelloService.dto()
	at java.base@21.0.1/java.lang.Class.checkMethod(DynamicHub.java:1065)
	at java.base@21.0.1/java.lang.Class.getMethod(DynamicHub.java:1050)
	at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getMethod(ResteasyReactiveResourceInfo.java:79)
```

### Expected behavior

Endpoint class should be found.

### Actual behavior

See description.

### How to Reproduce?

Use this repo / branch, see README
* https://github.com/alesj/quarkus_cl130/tree/nativerest1

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

3.13.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

https://gist.github.com/alesj/7bc3f4737a2f32c4dcaa98bf3c5543cd

If you add @RegisterForReflection to REST endpoint class, it works.

The sample that @alesj provided indeed does not work with 3.12. 
My usecase was a little different and was fixed with downgrading to 3.12.
In JVM mode it works with latest version as well.

My filter:
```
package com.xyz

import jakarta.annotation.Priority
import jakarta.ws.rs.Priorities.AUTHENTICATION
import jakarta.ws.rs.container.ContainerRequestContext
import jakarta.ws.rs.container.ContainerRequestFilter
import jakarta.ws.rs.container.ResourceInfo
import jakarta.ws.rs.core.Context
import jakarta.ws.rs.core.Response
import jakarta.ws.rs.core.Response.Status.UNAVAILABLE_FOR_LEGAL_REASONS
import jakarta.ws.rs.ext.Provider

@Provider
@Priority(AUTHENTICATION)
@AllowNoProfile
class ProfileFilter : ContainerRequestFilter {

  @Context
  lateinit var resourceInfo: ResourceInfo

  override fun filter(context: ContainerRequestContext) {
    val userUid = context.securityContext.userPrincipal?.name ?: return
    if (resourceInfo.resourceClass.getAnnotation(AllowNoProfile::class.java) != null ||
      resourceInfo.resourceMethod.getAnnotation(AllowNoProfile::class.java) != null
    ) return

    if (userUid === "foobar") {
      context.abortWith(Response.status(UNAVAILABLE_FOR_LEGAL_REASONS).build())
    }
  }
}
```

Stacktrace (removed sensitive parts, ProfileFilter line 33 and 43 are wrong - exception is thrown when accessing resourceMethod):
```
at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getMethod(ResteasyReactiveResourceInfo.java:84)
--
at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getResourceMethod(ResteasyReactiveResourceInfo.java:115)
at jakarta.ws.rs.container.ContextProducers_ProducerMethod_resourceInfo_O6EVLN_c9nIaKY9TO6cnD-zx-3c_ClientProxy.getResourceMethod(Unknown Source)
at com.xyz.ProfileFilter.hasAnnotation(ProfileFilter.kt:43)
at com.xyz.ProfileFilter.filter(ProfileFilter.kt:33)
at org.jboss.resteasy.reactive.server.handlers.ResourceRequestFilterHandler.handle(ResourceRequestFilterHandler.java:48)
at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:131)
at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:635)
at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
at java.base@21.0.4/java.lang.Thread.runWith(Thread.java:1596)
at java.base@21.0.4/java.lang.Thread.run(Thread.java:1583)
at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:896)
at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:872)

Caused by: java.lang.NoSuchMethodException: com.xyz.MyResource.getXyz(jakarta.ws.rs.core.SecurityContext, int)
at java.base@21.0.4/java.lang.Class.checkMethod(DynamicHub.java:1078)
at java.base@21.0.4/java.lang.Class.getMethod(DynamicHub.java:1063)
at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getMethod(ResteasyReactiveResourceInfo.java:79)
... 18 more
```

--- a/extensions/resteasy-reactive/rest/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java
+++ b/extensions/resteasy-reactive/rest/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java
@@ -1141,6 +1141,7 @@ public class ResteasyReactiveProcessor {
             SetupEndpointsResultBuildItem setupEndpointsResult,
             List<MessageBodyReaderBuildItem> messageBodyReaderBuildItems,
             List<MessageBodyWriterBuildItem> messageBodyWriterBuildItems,
+            ResourceInterceptorsBuildItem resourceInterceptorsBuildItem,
             BuildProducer<ReflectiveClassBuildItem> producer) {
         List<ResourceClass> resourceClasses = setupEndpointsResult.getResourceClasses();
         IndexView index = beanArchiveIndexBuildItem.getIndex();
@@ -1185,6 +1186,13 @@ public class ResteasyReactiveProcessor {
                 }
             }
         }
+
+        // when a ContainerResponseFilter exists, it can potentially do responseContext.setEntityStream()
+        // which then forces the use of the slow path for calling writers
+        if (!resourceInterceptorsBuildItem.getResourceInterceptors().getContainerResponseFilters().isEmpty()) {
+            serializersRequireResourceReflection = true;
+        }
+
         if (serializersRequireResourceReflection) {
             producer.produce(ReflectiveClassBuildItem
                     .builder(resourceClasses.stream().map(ResourceClass::getClassName).toArray(String[]::new))

--- a/independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/model/InterceptorContainer.java
+++ b/independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/model/InterceptorContainer.java
@@ -33,6 +33,10 @@ public class InterceptorContainer<T> {
         Collections.sort(nameResourceInterceptors);
     }
 
+    public boolean isEmpty() {
+        return globalResourceInterceptors.isEmpty() && nameResourceInterceptors.isEmpty();
+    }
+
     public ResourceInterceptor<T> create() {
         return new ResourceInterceptor<>();
     }

--- a/integration-tests/resteasy-reactive-kotlin/standard/src/main/kotlin/io/quarkus/it/resteasy/reactive/kotlin/EntityStreamSettingContainerResponseFilter.kt
+++ b/integration-tests/resteasy-reactive-kotlin/standard/src/main/kotlin/io/quarkus/it/resteasy/reactive/kotlin/EntityStreamSettingContainerResponseFilter.kt
@@ -1,0 +1,28 @@
+package io.quarkus.it.resteasy.reactive.kotlin
+
+import jakarta.ws.rs.container.ContainerRequestContext
+import jakarta.ws.rs.container.ContainerResponseContext
+import jakarta.ws.rs.container.ContainerResponseFilter
+import jakarta.ws.rs.ext.Provider
+import java.io.ByteArrayInputStream
+import org.jboss.resteasy.reactive.server.spi.ResteasyReactiveContainerRequestContext
+
+@Provider
+class EntityStreamSettingContainerResponseFilter : ContainerResponseFilter {
+    override fun filter(
+        requestContext: ContainerRequestContext?,
+        responseContext: ContainerResponseContext?
+    ) {
+        if (requestContext is ResteasyReactiveContainerRequestContext) {
+            if (
+                "hello".equals(
+                    requestContext.serverRequestContext.resteasyReactiveResourceInfo.name
+                )
+            ) {
+                responseContext?.setEntity(
+                    ByteArrayInputStream("Hello Quarkus REST".toByteArray(Charsets.UTF_8))
+                )
+            }
+        }
+    }
+}

--- a/integration-tests/resteasy-reactive-kotlin/standard/src/main/kotlin/io/quarkus/it/resteasy/reactive/kotlin/ReactiveGreetingResource.kt
+++ b/integration-tests/resteasy-reactive-kotlin/standard/src/main/kotlin/io/quarkus/it/resteasy/reactive/kotlin/ReactiveGreetingResource.kt
@@ -30,7 +30,7 @@ class ReactiveGreetingResource @Inject constructor(val req: RequestScopedKotlinC
     @GET
     @Produces(MediaType.TEXT_PLAIN)
     @Path("/{name}")
-    suspend fun hello(name: String): String {
+    suspend fun namedHello(name: String): String {
         delay(50)
         return "Hello $name"
     }

--- a/integration-tests/resteasy-reactive-kotlin/standard/src/test/kotlin/io/quarkus/it/resteasy/reactive/kotlin/ReactiveGreetingResourceTest.kt
+++ b/integration-tests/resteasy-reactive-kotlin/standard/src/test/kotlin/io/quarkus/it/resteasy/reactive/kotlin/ReactiveGreetingResourceTest.kt
@@ -19,7 +19,9 @@ class ReactiveGreetingResourceTest {
         When { get("/hello-resteasy-reactive/") } Then
             {
                 statusCode(200)
-                body(`is`("Hello RestEASY Reactive"))
+                body(
+                    `is`("Hello Quarkus REST")
+                ) // the result comes from EntityStreamSettingContainerResponseFilter
             }
     }
 

